presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PreferredProperties.java

```diff
@@ -130,9 +139,9 @@ public static PreferredProperties local(List<? extends LocalProperty<VariableRef
         return localProperties;
     }
 
-    public PreferredProperties mergeWithParent(PreferredProperties parent, boolean mergePartitionPreference, boolean useExactPartitioning)
+    public PreferredProperties mergeWithParent(PreferredProperties parent, MergeWithParentStrategy mergeWithParentStrategy)
```

Comment:
From what i understand there are only 2 differences between `EXACT` and `MERGE_PARTITION_PREFERENCE`: when `EXACT` is chosen - the local properties are not merged, and if the current property requirement has no partitioning requirement - parent partitioning requirement is taken. 

The `localProperties` in the `PreferredProperties` (and expecially merging them) makes no sense. So I checked how the `localProperties` properties are used, and I have found that the `localProperties` are effectively unused. Thus I wouldn't bother about the `localProperties` for now. Ideally it would be nice to have a refactoring PR, and completely remove this field to avoid confusion in the future.

For the "no partitioning" requirement case - I don't think there's any practical difference.

Thus, I'm trying to imply that we should simply have a boolean flag here.